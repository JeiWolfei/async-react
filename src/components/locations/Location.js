import React from 'react';
import PropTypes from 'prop-types';

function Location({ location }) {
  const { name, type, dimension, created, residents } = location;
  return (
    <ul style={{ listStyle: 'none' }}>
      <a href={location.url} target={'_blank'}>
        <li style={{
          fontWeight: 'bold',
          color: 'rgb(60, 180, 190)',
          textDecoration: 'underline'
        }}>
          {name}
        </li>
      </a>
      <li>{type}</li>
      <li>{dimension}</li>
      <li>
        {new Date(created)
          .toUTCString()
          .split(' ')
          .slice(0, 4)
          .join(' ')}
      </li>
      <li>Residents: {residents.length}</li>
    </ul>
  );
}

Location.propTypes = {
  location: PropTypes.object.isRequired
};

export default Location;
