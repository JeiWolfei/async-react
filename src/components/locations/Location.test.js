import React from 'react';
import renderer from 'react-test-renderer';
import Location from './Location';

describe('Location', () => {
  it('matches a snapshot', () => {
    const location = {
      name: 'Earth (C-137)',
      type: 'Planet',
      dimension: 'Dimension C-137',
      created: '2017-11-10T12:42:04.162Z',
      residents: '27'
    };
    const tree = renderer.create(
      <Location location={location}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

