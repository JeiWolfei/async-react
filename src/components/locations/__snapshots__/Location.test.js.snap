// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location matches a snapshot 1`] = `
<ul
  style={
    Object {
      "listStyle": "none",
    }
  }
>
  <a
    target="_blank"
  >
    <li
      style={
        Object {
          "color": "rgb(60, 180, 190)",
          "fontWeight": "bold",
          "textDecoration": "underline",
        }
      }
    >
      Earth (C-137)
    </li>
  </a>
  <li>
    Type: 
    Planet
  </li>
  <li>
    Dimension: 
    Dimension C-137
  </li>
  <li>
    Created: 
    Fri, 10 Nov 2017
  </li>
  <li>
    Residents: 
    2
  </li>
</ul>
`;
